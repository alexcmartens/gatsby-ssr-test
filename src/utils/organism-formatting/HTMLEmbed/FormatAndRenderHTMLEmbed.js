import React from "react"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import { HTMLEmbed } from "terra-one"

const FormatAndRenderHTMLEmbed = ({ data, paddingClass, index, location }) => {
  const header = data?.header
  const subheader = data?.subHeader
  const HTML = data?.html?.blob?.childMarkdownRemark?.html
  const ratio = data?.componentAspectRatio
  const content = {
    header,
    subheader,
    HTML,
    ratio,
  }
  return (
    <>{renderPaddingOrFullBleed(<HTMLEmbed content={content} key={index} />)}</>
  )
}

export default FormatAndRenderHTMLEmbed
