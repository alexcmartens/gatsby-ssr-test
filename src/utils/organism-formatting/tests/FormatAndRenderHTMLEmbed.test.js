import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderHTMLEmbed from "../HTMLEmbed/FormatAndRenderHTMLEmbed"

describe("FormatAndRenderHTMLEmbed", () => {
  it("renders correctly", () => {
    const html = "<div><p>I AM HTML!</p></div>"

    const data = { html }

    const tree = renderer
      .create(<FormatAndRenderHTMLEmbed data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
