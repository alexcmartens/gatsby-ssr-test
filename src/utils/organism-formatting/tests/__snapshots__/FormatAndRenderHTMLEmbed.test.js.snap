// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderHTMLEmbed renders correctly 1`] = `
<section
  className="default-layout   mockconstructor
      "
  style={
    Object {
      "backgroundColor": undefined,
      "marginBottom": "",
    }
  }
>
  <section
    className="default-layout__grid 
          
        "
    style={
      Object {
        "backgroundColor": "",
        "marginBottom": "",
      }
    }
  >
    <div
      className="default-layout__children "
    >
      <div
        content={
          Object {
            "HTML": undefined,
            "header": undefined,
            "ratio": undefined,
            "subheader": undefined,
          }
        }
      />
    </div>
  </section>
</section>
`;
